contracts/NFTAttachment.sol

```diff
@@ -77,6 +84,21 @@ contract NFTAttachment is /* AccessControl, */ MetaTxContext {
         emit MinterAttachToNFT(minter, nftId, attachment);
     }
 
+    /** Attach data `attachment` to the NFT type `nftId`, as manager of collection.
+     *
+     * This only works with NFTs on Collections.
+     */
+    function collectionManagerAttachToNFT(uint256 nftId, bytes calldata attachment)
+    public {
+        (address collection, uint32 innerId, uint64 zeroSupply) = _parseNftId(nftId);
+        require(zeroSupply == 0, "NFT format is not compatible with Collections");
+
+        address manager = _msgSender();
+        require(Collection(collection).hasRole(COLLECTION_MANAGER_ROLE, manager), "sender is not collection manager");
```

Comment:
why do we have it? 
Is security workaround? 

Didn't get how should it help, there is still an issue - anyone with this role can change content of others NFTs, isn't it?

Reply:
It's not workaround, this is implementation that looks solid.
Collection manager in the owner of the collection who can operate on it, so he is responsible for asset attachments

Reply:
> require(Collection(collection).hasRole(COLLECTION_MANAGER_ROLE, manager), "sender is not collection manager")

does this line check that specific wallet has role on specific collection? can't check hasRole method behaviour via side

Reply:
Yes.

Reply:
The function is in the dependency `AccessControl`